Lib/http/server.py

```diff
@@ -686,12 +687,42 @@ def send_head(self):
         except OSError:
             self.send_error(HTTPStatus.NOT_FOUND, "File not found")
             return None
+
         try:
+            fs = os.fstat(f.fileno())
+            # Use browser cache if possible
+            if ("If-Modified-Since" in self.headers
+                    and "If-None-Match" not in self.headers):
+                # compare If-Modified-Since and time of last file modification
+                try:
+                    ims = email.utils.parsedate_to_datetime(
+                        self.headers["If-Modified-Since"])
+                except:
```

Comment:
That TypeError looks like a bug in the implementation. Other exceptions:

>>> parsedate_to_datetime("Sun, 99 Feb 2017 20:49:31")
ValueError: day is out of range for month
>>> parsedate_to_datetime("Sun, 26 Feb 2017 20:49:31 +12345678901234")
OverflowError: Python int too large to convert to C long
>>> parsedate_to_datetime("Mon, 20 Nov , 19:12:08 GMT")
IndexError: string index out of range

The point is the outer exception handler reraises the exception, but your handler swallows it.